### PR TITLE
scale-test: Scrape APIServer only metrics

### PR DESCRIPTION
**Pull request by hakuna-matatah**

### CODE REVIEW

**Comment by hakuna-matatah** [comment]
Given the small scale load test is succeeding consistently and large scale isn't due to following error :
```
>���Failure3no endpoints available for service "prometheus-k8s""ServiceUnavailable0��"�
F1020 03:14:27.263151   13549 clusterloader.go:326] Error while setting up prometheus stack: timed out waiting for the condition
```
As we can debug the issue due to access issues, we have to take an educated guess for now, so, it makes me think that Prometheus pod is not able to come up due to the memory requirements and given we provide `[large ec2 instance type](https://github.com/kubernetes/perf-tests/blob/master/clusterloader2/pkg/prometheus/manifests/prometheus-prometheus.yaml#L33-L35) which has 4G memory`, 10G is not available for prometheus pod as per the calculation setting mentioned here - https://github.com/kubernetes/perf-tests/blob/master/clusterloader2/pkg/prometheus/manifests/prometheus-prometheus.yaml#L40-L41

**Comment by hakuna-matatah** [comment]
Last test failed due to unavailability of EC2 nodes

````
InstanceGroup	nodes-us-east-2b	InstanceGroup "nodes-us-east-2b" did not have enough nodes 800 vs 1667
InstanceGroup	nodes-us-east-2c	InstanceGroup "nodes-us-east-2c" did not have enough nodes 891 vs 1666
```

**Comment by hakuna-matatah** [comment]
Same error as above, ec2 instance types doesn't seem to be available, let me add more to the mixed fleet.
```
InstanceGroup	nodes-us-east-2b	InstanceGroup "nodes-us-east-2b" did not have enough nodes 967 vs 1667
InstanceGroup	nodes-us-east-2c	InstanceGroup "nodes-us-east-2c" did not have enough nodes 950 vs 1666```

File: tests/e2e/scenarios/scalability/run-test.sh [hunk 1]
```diff
@@ -163,6 +163,8 @@ export KUBE_SSH_KEY_PATH="/tmp/kops/${CLUSTER_NAME}/id_ed25519"
 # this is used as a label to select kube-proxy pods on kops for kube-proxy service 
 # used by CL2 Prometheus here https://github.com/kubernetes/perf-tests/blob/master/clusterloader2/pkg/prometheus/manifests/default/kube-proxy-service.yaml#L2
 export PROMETHEUS_KUBE_PROXY_SELECTOR_KEY="k8s-app"
+export PROMETHEUS_SCRAPE_APISERVER_ONLY="true"
+export CL2_PROMETHEUS_NODE_SELECTOR="node-role.kubernetes.io/control-plane: "
```
Comment (hakman) [inline tests/e2e/scenarios/scalability/run-test.sh:167]:
```suggestion
export CL2_PROMETHEUS_NODE_SELECTOR="node-role.kubernetes.io/control-plane: "
export CL2_PROMETHEUS_TOLERATE_MASTER="true"
```

File: tests/e2e/scenarios/scalability/run-test.sh [hunk 2]
```diff
@@ -163,6 +163,9 @@ export KUBE_SSH_KEY_PATH="/tmp/kops/${CLUSTER_NAME}/id_ed25519"
...
```
Comment (hakman) [inline tests/e2e/scenarios/scalability/run-test.sh:167]:
```suggestion
```

**Comment by hakuna-matatah** [comment]
```
Your cluster e2e-ff02749ef8-a423a.test-cncf-aws.k8s.io is ready
I1025 17:08:27.065287   13030 validate_cluster.go:220] (will retry): cluster passed validation 8 consecutive times
Error: validation failed: wait time exceeded during validation
Error: exit status 1
```
Looks like the wait time exceeded even though cluster passed validation 8 consecutive times. Will need to increase the validation time or reduce the number from `10` to `5` for consecutive successful validation checks.

**Comment by hakuna-matatah** [comment]
//test presubmit-kops-aws-small-scale-amazonvpc-using-cl2

**Comment by hakuna-matatah** [comment]
Looks like we are experiencing throttling during the creation of cluster  in the latest run 

```
W1025 18:59:46.369792   12985 executor.go:140] error running task "AutoscalingGroup/nodes-us-east-2c.e2e-ff02749ef8-a423a.test-cncf-aws.k8s.io" (-12m11s remaining to succeed): error listing AutoscalingGroups: Throttling: Rate exceeded
	status code: 400, request id: 50cc0b52-fcb6-4845-9af3-a6865416f6d0
Error: error running tasks: deadline exceeded executing task AutoscalingGroup/nodes-us-east-2c.e2e-ff02749ef8-a423a.test-cncf-aws.k8s.io. Example error: error listing AutoscalingGroups: Throttling: Rate exceeded
	status code: 400, request id: 50cc0b52-fcb6-4845-9af3-a6865416f6d0
Error: exit status 1
```

**Review by hakman — APPROVED** [review]

**Pull request merged**